pkg/ccl/changefeedccl/changefeed_stmt.go

```diff
@@ -1102,6 +1109,8 @@ func (b *changefeedResumer) OnPauseRequest(
 	cp := progress.GetChangefeed()
 	execCfg := jobExec.(sql.JobExecContext).ExecCfg()
 
+	b.updatePausedMetrics(ctx, jobExec)
```

Comment:
I suspect this will not work as intended.
This counts the number of times attempt to pause the job happens; It doesn't quite tell you how many changefeed jobs are paused right now.

Also, consider that this function (OnPauseRequested) maybe executed multiple times (e.g. txn restart).

I suspect a place to do this would be to have job registry keep track of some sort of functional gauge; somewhere  around https://github.com/cockroachdb/cockroach/blob/master/pkg/jobs/registry.go#L745-L745
pauseAndCancel can return the number of jobs successfully paused and cancelled on that node; that result can be made into functional per job type gauge; and that gauge value can be summed across cluster.

Reply:
I pushed some updates. Finding where we set the status to `Paused` was easy. I think the hard part is decrementing the Gauge when we unpause a job. Due to intermediate states like `reverting` etc., it's hard to tell the previous state when you cancel/resume a job. You usually have to look two states back. I ended up putting some logic in `Job.update`.